internal/services/paloalto/next_generation_firewall_vhub_panorama_resource.go

```diff
@@ -0,0 +1,264 @@
+					return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+				}
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			firewall := firewalls.FirewallResource{
+				Location: location.Normalize(model.Location),
+				Properties: firewalls.FirewallDeploymentProperties{
+					PanoramaConfig: &firewalls.PanoramaConfig{
+						ConfigString: model.PanoramaBase64Config,
+					},
+					IsPanoramaManaged: pointer.To(firewalls.BooleanEnumTRUE),
+					DnsSettings:       schema.ExpandDNSSettings(model.DNSSettings),
+					MarketplaceDetails: firewalls.MarketplaceDetails{
+						OfferId:     "pan_swfw_cloud_ngfw", // TODO - Will just supplying the offer ID `panw-cloud-ngfw-payg` work?
+						PublisherId: "paloaltonetworks",
+					},
+					NetworkProfile: schema.ExpandNetworkProfileVHub(model.NetworkProfile),
+					PlanData: firewalls.PlanData{
+						BillingCycle: firewalls.BillingCycleMONTHLY,
+						PlanId:       "panw-cloud-ngfw-payg",
+					},
+					FrontEndSettings: schema.ExpandDestinationNAT(model.FrontEnd),
+				},
+				Tags: tags.Expand(model.Tags),
+			}
+
+			if err = client.CreateOrUpdateThenPoll(ctx, id, firewall); err != nil {
+				return err
+			}
+
+			metadata.SetID(id)
+
+			return nil
+		},
+	}
+}
+
+func (r NextGenerationFirewallVHubPanoramaResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.PaloAlto.FirewallClient
+
+			id, err := firewalls.ParseFirewallID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var state NextGenerationFirewallVHubPanoramaModel
+
+			existing, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(existing.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("reading %s: %+v", *id, err)
+			}
+
+			state.Name = id.FirewallName
+			state.ResourceGroupName = id.ResourceGroupName
+
+			props := existing.Model.Properties
+			state.DNSSettings = schema.FlattenDNSSettings(props.DnsSettings)
+
+			netProfile, err := schema.FlattenNetworkProfileVHub(props.NetworkProfile)
+			if err != nil {
+				return fmt.Errorf("parsing Network Profile for %s: %+v", *id, err)
+			}
+
+			state.NetworkProfile = []schema.NetworkProfileVHub{*netProfile}
+
+			if feSettings := pointer.From(props.FrontEndSettings); len(feSettings) != 0 {
+				fes := make([]schema.DestinationNAT, 0)
+				for _, v := range feSettings {
+					bePort, _ := strconv.Atoi(v.BackendConfiguration.Port)
+					fePort, _ := strconv.Atoi(v.FrontendConfiguration.Port)
+					fe := schema.DestinationNAT{
+						Name:     v.Name,
+						Protocol: string(v.Protocol),
+						BackendConfiguration: []schema.BackendEndpointConfiguration{{
+							PublicIP: pointer.From(v.BackendConfiguration.Address.Address),
```

Comment:
confirming, is `v.BackendConfiguration` or `v.BackendConfiguration.Address` a pointer?

Reply:
No, neither are pointers.